Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -475,6 +476,18 @@ DirectoryBlobReader::DirectoryBlobReader(
   }
 }
 
+DirectoryBlobReader::DirectoryBlobReader(const DirectoryBlobReader& rhs)
+    : m_gamecube_pseudopartition(rhs.m_gamecube_pseudopartition),
+      m_nonpartition_contents(rhs.m_nonpartition_contents), m_encryption_cache(this),
+      m_is_wii(rhs.m_is_wii), m_encrypted(rhs.m_encrypted), m_data_size(rhs.m_data_size),
+      m_wrapped_volume(CreateDisc(rhs.m_wrapped_volume->GetBlobReader().CopyReader()))
```

Comment:
I tried running the code, and I got a crash here. I think the solution would be something like:

```suggestion
      m_wrapped_volume(rhs.m_wrapped_volume ? CreateDisc(rhs.m_wrapped_volume->GetBlobReader().CopyReader()) : nullptr)
```

I would recommend testing your `DirectoryBlob` changes for yourself. If you want a simple way to ensure `CopyReader` gets called, you can replace the implementation of `CreateVolume` in `Volume.cpp` with `return CreateVolume(CreateBlobReader(path)->CopyReader());`. Then launch a game or open the game properties.

Reply:
Or actually, do we need `m_wrapped_volume`? I can't find anything it gets used for after the constructor finishes running.

Reply:
SetPartitionHeader uses it a few times.

Reply:
Yes, but that function is only used while the constructor is running.